### PR TITLE
add test_volume_reattach_after_engine_sigkill

### PR DESCRIPTION
**Pull request by meldafrawi**
longhorn/longhorn#1240
Signed-off-by: Mohamed Eldafrawi <mohamed.eldafrawi@rancher.com>

### CODE REVIEW

File: manager/integration/tests/common.py [hunk 1]
```diff
@@ -3337,3 +3337,38 @@ def settings_reset():
 
     client = get_longhorn_api_client()
     reset_settings(client)
+
+
+def crash_engine_process(client, core_api, volume_name):
```
Comment (yasker) [inline manager/integration/tests/common.py:3521]: `crash_engine_process_with_sigkill`

File: manager/integration/tests/test_ha.py [hunk 2]
```diff
@@ -1295,7 +1296,29 @@ def test_volume_reattach_after_engine_sigkill():  # NOQA
...
+    wait_for_volume_detached(client, volume_name)
+    wait_for_volume_healthy(client, volume_name)
+
+    wait_for_pod_restart(core_api, pod_name)
+    wait_for_pod_remount(core_api, pod_name)
+
+    res_md5sum = get_pod_data_md5sum(core_api, pod_name, data_path1)
+    assert md5sum == res_md5sum
+
+    write_pod_volume_random_data(core_api, pod_name,
```
Comment (yasker) [inline manager/integration/tests/test_ha.py:1320]: just add a final check to make sure the data was correct for the last write.

Comment (meldafrawi) [reply manager/integration/tests/test_ha.py:1320]: fixed

File: manager/integration/tests/common.py (hunk 1 above)
Comment (meldafrawi) [reply manager/integration/tests/common.py]: fixed

**Comment by yasker** [comment]
flake8 failed.

**Comment by meldafrawi** [comment]
flake8 fixed

**Pull request merged**